src/ssl.jl

```diff
@@ -434,6 +433,7 @@ check_isopen(ssl::SSLStream, op) = isopen(ssl) || throw(Base.IOError("$op requir
 
 macro geterror(expr)
```

Comment:
let's change this to return a value, rather than overwrite variables names `ret` (and `err` and `ssl`)

maybe something like:
```suggestion
macro geterror(ssl, expr)
    quote
        clear_errors!()
        ret = $(esc(expr))
        if ret <= 0
            ssl = $(esc(ssl))
            err = get_error(ssl.ssl, ret)
            if err == SSL_ERROR_ZERO_RETURN
                @atomicset ssl.close_notify_received = true
            elseif err == SSL_ERROR_NONE
                # pass
            elseif err == SSL_ERROR_WANT_READ
                ret = SSL_ERROR_WANT_READ
            elseif err == SSL_ERROR_WANT_WRITE
                ret = SSL_ERROR_WANT_WRITE
            else
                close(ssl, false)
                throw(Base.IOError(OpenSSLError(err).msg, 0))
            end
        end
        ret
    end
end
```

Also why are we doing `Base.IOError(OpenSSLError(err).msg, 0)` rather than just `Base.IOError(err, 0)`?


(And why is it so hard to leave multi-line suggestion in GitHub?!)

Reply:
or if we also want to always return a enum value, something like
```julia
macro geterror(ssl, expr)
    quote
        clear_errors!()
        ret = $(esc(expr))
        if ret == 1
            ret = SSL_ERROR_NONE
        else
            ssl = $(esc(ssl))
            err = get_error(ssl.ssl, ret)
            if err == SSL_ERROR_ZERO_RETURN
                @atomicset ssl.close_notify_received = true
            elseif err == SSL_ERROR_NONE
                # pass
            elseif err == SSL_ERROR_WANT_READ
                ret = SSL_ERROR_WANT_READ
            elseif err == SSL_ERROR_WANT_WRITE
                ret = SSL_ERROR_WANT_WRITE
            else
                close(ssl, false)
                throw(Base.IOError(OpenSSLError(err).msg, 0))
            end
        end
        ret
    end
end